prov/verbs/src/verbs_msg.c

```diff
@@ -47,6 +47,11 @@ fi_ibv_msg_ep_recvmsg(struct fid_ep *ep, const struct fi_msg *msg, uint64_t flag
 	_ep = container_of(ep, struct fi_ibv_msg_ep, ep_fid);
 	assert(_ep->rcq);
 
+	if (_ep->srq_ep) {
+		/* Cannot post a recv WR to a QP where a SRQ is attached to */
+		return -FI_ENOSYS;
+	}
```

Comment:
Can we adjust the code to set the recv function pointer to the enosys version if a shared rx context is in use, and avoid adding this check in the fast path for every post recv operation?